qt-models/divetripmodel.cpp

```diff
@@ -1106,3 +1060,84 @@ void DiveTripModel::filterFinished()
 		dataChanged(tripIndex, tripIndex);
 	}
 }
+
+// Simple sorting helper for sorting against a criterium and if
+// that is undefined against a different criterium.
+// Return true if diff1 < 0, false if diff1 > 0.
+// If diff1 == 0 return true if diff2 < 0;
+static bool lessThanHelper(int diff1, int diff2)
+{
+	return diff1 < 0 || (diff1 == 0 && diff2 < 0);
+}
+
+static int strCmp(const char *s1, const char *s2)
+{
+	if (!s1)
+		return !s2 ? 0 : -1;
+	if (!s2)
+		return 1;
+	return strcasecmp(s1, s2); // TODO: support for non-ASCII UTF-8!
```

Comment:
yep, strcasecmp cannot be used for UTF-8. QString already has this support.

Reply:
Ok, so let's go via `QString` at first and then see if we can avoid the unnecessary copying by using `std::collate` or something. But of course this makes only sense with benchmarking.

Today I will not find time to work on this, hopefully tomorrow.